### ISSUE TITLE
[SAE] AssignmentMode::JumpReLU is a bounded sigmoid gate, not magnitude-carrying JumpReLU

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#598]
**Verified — CONFIRMED.**

The enum is `JumpReLU { temperature, threshold }` (~`src/terms/sae_manifold.rs:1699`) but the implementation returns a bounded sigmoid (~5805-5806):
```rust
if logits[i] > threshold {
    out[i] = stable_logistic(logits[i] / temperature);   // ∈ (0,1)
}
```
(same form at ~5860-5868, 5917-5933, 5985-5997, 6058-6067).

The Python/torch and PyFFI docs define JumpReLU as **magnitude-carrying**: `φ(z) = z · 1[z > τ]` (`gamfit/torch/penalties.py:524,582`; PyFFI doc). These are not equivalent:
- z=10, θ=0, τ=1 → correct JumpReLU = 10; Rust = sigmoid(10) ≈ 0.99995.

So the Rust mode cannot carry activation magnitude through the assignment channel — it's a Bernoulli-style gate. This also diverges from JumpReLU SAEs (discontinuous, STE-trained for L0).

**Fix — pick one and be honest:**
- (A) Rename to `HardGatedSigmoid` / `BernoulliSigmoidGate` and stop calling it JumpReLU; **or**
- (B) Implement true JumpReLU `if z>θ {z} else {0}` with a centered surrogate derivative for optimization (see #threshold-centering issue) and L0-surrogate penalty.

Gated-SAE (separate gate × softplus magnitude) is a distinct family — don't conflate.

_Source: static SAE/manifold-SAE audit; verified against current code at commit 3c1a4efa2._

**Comment by SauersML** [comment]
Resolved by design. The JumpReLU assignment is deliberately a bounded gate in [0,1]; reconstruction MAGNITUDE is carried by the decoder curve φ(t)ᵀB_k, not the gate (this is the gate×decoder-magnitude factorization, documented at sae_manifold.rs:1722-1729). The original bug — an uncentered surrogate whose hard/smooth transitions disagreed — is fixed in #599. So the gate is intentionally not `z·1[z>τ]`; magnitude lives downstream.
